packages/flutter/lib/src/material/segmented_button_theme.dart

```diff
@@ -0,0 +1,140 @@
+  ///
+  /// Non-null properties or non-null resolved [MaterialStateProperty]
+  /// values override the default values used by [SegmentedButton].
+  ///
+  /// If [style] is null, then this theme doesn't override anything.
+  final ButtonStyle? style;
+
+  /// Override for [SegmentedButton.selectedIcon].
+  ///
+  /// If non-null then [selectedIcon] will be used instead of default
+  /// value for [Segmented.selectedIcon].
+  final Widget? selectedIcon;
+
+  /// Creates a copy of this object but with the given fields replaced with the
+  /// new values.
+  SegmentedButtonThemeData copyWith({
+    ButtonStyle? style,
+    Widget? selectedIcon,
+  }) {
+    return SegmentedButtonThemeData(
+      style: style ?? this.style,
+      selectedIcon: selectedIcon ?? this.selectedIcon,
+    );
+  }
+
+  /// Linearly interpolate between two segmented button themes.
+  static SegmentedButtonThemeData lerp(SegmentedButtonThemeData? a, SegmentedButtonThemeData? b, double t) {
+    return SegmentedButtonThemeData(
+      style: ButtonStyle.lerp(a?.style, b?.style, t),
+      selectedIcon: t < 0.5 ? a?.selectedIcon : b?.selectedIcon,
+    );
+  }
+
+  @override
+  int get hashCode => Object.hash(
+    style,
+    selectedIcon,
+  );
+
+  @override
+  bool operator ==(Object other) {
+    if (identical(this, other)) {
+      return true;
+    }
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
+    return other is SegmentedButtonThemeData
+        && other.style == style
+        && other.selectedIcon == selectedIcon;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<ButtonStyle>('style', style, defaultValue: null));
+    properties.add(DiagnosticsProperty<Widget>('selectedIcon', selectedIcon, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the visual properties for
+/// [SegmentedButton]s in this widget's subtree.
+///
+/// Values specified here are used for [SegmentedButton] properties that are not
+/// given an explicit non-null value.
+class SegmentedButtonTheme extends InheritedTheme {
+  /// Creates a segmented button theme that controls visual parameters for
+  /// descendent [SegmentedButton]s.
+  const SegmentedButtonTheme({
+    super.key,
+    required this.data,
+    required super.child,
+  }) : assert(data != null);
+
+  /// Specifies the visual properties used by descendant [SegmentedButton]
+  /// widgets.
+  final SegmentedButtonThemeData data;
+
+  /// The closest instance of this class that encloses the given context.
+  ///
+  /// If there is no enclosing [SegmentedButtonTheme] widget, then
+  /// [ThemeData.segmentedButtonTheme] is used.
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// SegmentedButtonThemeData theme = SegmentedButtonTheme.of(context);
+  /// ```
+  static SegmentedButtonThemeData of(BuildContext context) {
```

Comment:
There should also be a related `maybeOf` for this static that returns a nullable `SegmentedButtonThemeData`. You can then implement this function in terms of that one.

Also take a look at the doc comments for the other `of` functions: you should include what happens if nothing is found, and a "See also" section that points to the `maybeOf` (and vice versa).

Is this also idiomatic with other button themes?  It seems odd that it would return the `Theme.of(context)` version if it didn't find one, instead of having the user do that fallback using `maybeOf`. I would have expected it to throw.

Reply:
This was pretty much lifted from TextButtonThemeData.of. I will update this to match other `of` functions. We can fix the others in a separate PR.

Reply:
And yes it is idiomatic of all the themes to fallback to the `Theme.of(context).fooTheme` in the `.of` method.